shims/spark310/src/main/scala/com/nvidia/spark/rapids/shims/spark310/ParquetCachedBatchSerializer.scala

```diff
@@ -359,44 +372,57 @@ class ParquetCachedBatchSerializer extends CachedBatchSerializer with Arm {
     cbRdd
   }
 
-  private def getColumnIndices(selectedAttributes: Seq[Attribute],
-     cacheAttributes: Seq[Attribute]): Seq[Int] = {
-    selectedAttributes.map(a => cacheAttributes.map(_.exprId).indexOf(a.exprId))
+  private def getColumnNames(
+     selectedAttributes: Seq[Attribute],
+     cacheAttributes: Seq[Attribute]): Seq[String] = {
+    selectedAttributes.map(a => "_col" + cacheAttributes.map(_.exprId).indexOf(a.exprId))
   }
 
   /**
    * Convert the cached data into a ColumnarBatch taking the result data back to the host
    *
-   * @param input the cached batches that should be converted.
-   * @param cacheAttributes the attributes of the data in the batch.
+   * @param input              the cached batches that should be converted.
+   * @param cacheAttributes    the attributes of the data in the batch.
    * @param selectedAttributes the fields that should be loaded from the data and the order they
    *                           should appear in the output batch.
-   * @param conf the configuration for the job.
+   * @param conf               the configuration for the job.
    * @return an RDD of the input cached batches transformed into the ColumnarBatch format.
    */
   override def convertCachedBatchToColumnarBatch(input: RDD[CachedBatch],
      cacheAttributes: Seq[Attribute],
      selectedAttributes: Seq[Attribute],
      conf: SQLConf): RDD[ColumnarBatch] = {
-    val batches = convertCachedBatchToColumnarInternal(input, cacheAttributes,
-      selectedAttributes)
-    val cbRdd = batches.map(batch => {
-      withResource(batch) { gpuBatch =>
-        val cols = GpuColumnVector.extractColumns(gpuBatch)
-        new ColumnarBatch(cols.map(_.copyToHost()).toArray, gpuBatch.numRows())
+    val rapidsConf = new RapidsConf(conf)
+    if (rapidsConf.isSqlEnabled) {
+      val batches = convertCachedBatchToColumnarInternal(input, cacheAttributes,
+        selectedAttributes)
+      val cbRdd = batches.map(batch => {
+        withResource(batch) { gpuBatch =>
+          val cols = GpuColumnVector.extractColumns(gpuBatch)
+          new ColumnarBatch(cols.map(_.copyToHost()).toArray, gpuBatch.numRows())
```

Comment:
This should use `safeMap` to avoid leaking host column vector instances if this throws during the mapping.